### PR TITLE
[W5.7][T16-3]Rahul Rajesh

### PR DESCRIPTION
**Pull request by rrtheonlyone**
Added Update Command

Still a work in progress

### CODE REVIEW

File: src/seedu/addressbook/commands/UpdateCommand.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import seedu.addressbook.data.person.Name;
+
+public class UpdateCommand extends Command {
+
+    public static final String COMMAND_WORD = "update";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Updates the person identified by the index number used in the last person listing.\n"
+            + "Parameters: INDEX\n"
+            + "Example: " + COMMAND_WORD + " 1";
```
Comment (juxd) [inline src/seedu/addressbook/commands/UpdateCommand.java]: shouldn't this be:
```
"Example: COMMAND_WORD [index] [new name]"
```
the new name isn't reflected here.

File: src/seedu/addressbook/parser/Parser.java [hunk 2]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (juxd) [inline src/seedu/addressbook/parser/Parser.java:14]: !!!

**Review by juxd — CHANGES_REQUESTED** [review]
Preliminary checks

**Comment by juxd** [comment]
CI fails because you did not update your I/O tests. Please check again.

File: src/seedu/addressbook/parser/Parser.java (hunk 2 above)
Comment (monmanuela) [reply src/seedu/addressbook/parser/Parser.java:14]: "While this is more readable, this is bad practice as it would affect performance and potentially pollute the namespace. `seedu.addressbook.commands` is a big library!"
Quoted from @juxd

Comment (rrtheonlyone) [reply src/seedu/addressbook/parser/Parser.java:14]: Hi guys,

If you look at `seedu.addressbook.commands`, `Parser.java` actually uses almost ALL the commands inside. So it does not pollute the namespace here. Plus, IntelliJ optimised this for me... 

@monmanuela `java.util` is not relevant here

Cheers!

File: src/seedu/addressbook/commands/UpdateCommand.java (hunk 1 above)
Comment (rrtheonlyone) [reply src/seedu/addressbook/commands/UpdateCommand.java]: Added in!

**Review by rrtheonlyone — COMMENTED** [review]
I have worked on some changes and fixed the tests!

**Comment by juxd** [comment]
LGTM

**Review by juxd — APPROVED** [review]

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 3]
```diff
@@ -10,6 +10,20 @@
 
 public class UtilsTest {
 
+    @Test
```
Comment (tanhengyeow) [inline test/java/seedu/addressbook/common/UtilsTest.java:13]: Please avoid unrelated changes in a new PR. In this case, I'm aware that this is work for the Week 4 LO. However, we strive for clean PRs (only related changes in one PR).

Clean PRs would facilitate easier code reviews for larger scale PRs, which would happen soon when your team is doing code reviews of each others' work in the project :)

File: src/seedu/addressbook/parser/Parser.java (hunk 2 above)
Comment (tanhengyeow) [reply src/seedu/addressbook/parser/Parser.java:14]: @rrtheonlyone Although you're right that `Parser.java` actually uses almost all the commands inside, it is still not a good practice to do a wildcard import. 

Let us look at this issue in the perspective of these 2 scenarios:
1. `seedu.addressbook.commands` potentially having more commands in the future
2. `Parser.java` not requiring some of the commands that are imported now

In any case, on top of the issues addressed by your teammates, doing a wildcard import would also affect maintainability. Therefore, it is still good practice to only import what you need :)

**Review by tanhengyeow — APPROVED** [review]
Good attempt! Great to see peer reviews too :) Refer to my comments below and close the PR after acknowledging the comments 😄

**Comment by rrtheonlyone** [comment]
Thanks for the feedback!
